### PR TITLE
Feature/auto shard rebalance

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose
The purpose of this PR is to integrate the [autorebalance](https://github.com/arangodb/autorebalance) tool into arangodb. 

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document: https://github.com/arangodb/documents/blob/master/DesignDocuments/01_IDEAS/AutoRebalance.md

### CODE REVIEW

File: Documentation/DocuBlocks/Rest/Administration/get_admin_cluster_rebalance.md [hunk 1]
```diff
@@ -4,20 +4,36 @@
 @RESTHEADER{GET /_admin/cluster/rebalanc, Calculates the current cluster imbalance}
 
 @RESTDESCRIPTION
-Calculates the current cluster imbalance and returns the result.
+Calculates the current cluster imbalance and returns the result. 
+It additionally shows the amount of ongoing and pending moveshard operations.
```
Comment (nerpaula) [inline Documentation/DocuBlocks/Rest/Administration/get_admin_cluster_rebalance.md:8]: please change to "pending move shard operations"

Comment (maierlars) [reply Documentation/DocuBlocks/Rest/Administration/get_admin_cluster_rebalance.md:8]:
```suggestion
It additionally shows the amount of ongoing and pending move shard operations.
```

File: arangod/RestHandler/RestAdminClusterHandler.cpp [hunk 2]
```diff
@@ -2135,6 +2143,413 @@ RestStatus RestAdminClusterHandler::handleRebalanceShards() {
...
+}
+
+RestStatus RestAdminClusterHandler::handleRebalanceExecute() {
+  if (request()->requestType() != rest::RequestType::POST) {
+    generateError(rest::ResponseCode::METHOD_NOT_ALLOWED,
+                  TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
+    return RestStatus::DONE;
+  }
+
+  auto batch = velocypack::deserialize<std::vector<MoveShardDescription>>(
```
Comment (neunhoef) [inline arangod/RestHandler/RestAdminClusterHandler.cpp:2232]: Lets make the API as follows: One sends a document with `moves` attribute being a list of moves, and set the `version` attribute to 1. Then we can extend the API a bit better. If we just get a list of moves, we have no wiggle room for further options.

Comment (maierlars) [reply arangod/RestHandler/RestAdminClusterHandler.cpp:2232]: ok.

File: Documentation/DocuBlocks/Rest/Administration/post_admin_cluster_rebalance_execute.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+@startDocuBlock post_admin_cluster_rebalance_execute
+@brief Execute the given set of move shard operations.
+
+@RESTHEADER{POST /_admin/cluster/rebalance/execute, Execute the given set of move shard operations}
+
+@RESTDESCRIPTION
+Execute the given set of move shard operations. Such a set can be calculated using
+other rebalance apis.
```
[the remark this answers is not in the document]
Comment (maierlars) [reply Documentation/DocuBlocks/Rest/Administration/post_admin_cluster_rebalance_execute.md:9]: done

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by maierlars** [comment]
```
* Test "restart"
    [FAILED]  tests\js\client\restart\test-soft-shutdown-cluster.js

      "test" failed: server unavailable for testing. Instance not healthy!
```
and
```
17:24:35 FATAL: command execution failed
17:24:35 java.io.EOFException
17:24:35 	at java.base/java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2905)
17:24:35 	at java.base/java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:3400)
17:24:35 	at java.base/java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:936)
17:24:35 	at java.base/java.io.ObjectInputStream.<init>(ObjectInputStream.java:379)
17:24:35 	at hudson.remoting.ObjectInputStreamEx.<init>(ObjectInputStreamEx.java:4
```

**Review by nerpaula — APPROVED** [review]

**Review by joerg84 — APPROVED** [review]

**Pull request merged**